### ISSUE TITLE
Compiler crash when use parenthesis in return

### ISSUE THREAD

**Issue by nagolove** [teal-language/tl#345]
Example
```lua
function foo(): number
    return (1, 1) as number
end
```
Report
```
/usr/bin/lua5.4: /home/testuser/.luarocks/share/lua/5.4/tl.lua:1723: attempt to index a nil value (local 'e1')
stack traceback:
        /home/testuser/.luarocks/share/lua/5.4/tl.lua:1723: in upvalue 'P'
        /home/testuser/.luarocks/share/lua/5.4/tl.lua:1759: in local 'parse_item'
        /home/testuser/.luarocks/share/lua/5.4/tl.lua:1279: in upvalue 'parse_trying_list'
        /home/testuser/.luarocks/share/lua/5.4/tl.lua:2306: in function </home/testuser/.luarocks/share/lua/5.4/tl.lua:2301>
        (...tail calls...)
        /home/testuser/.luarocks/share/lua/5.4/tl.lua:2479: in upvalue 'parse_statements'
        /home/testuser/.luarocks/share/lua/5.4/tl.lua:1507: in upvalue 'parse_function_args_rets_body'
        /home/testuser/.luarocks/share/lua/5.4/tl.lua:1898: in function </home/testuser/.luarocks/share/lua/5.4/tl.lua:1867>
        (...tail calls...)
        /home/testuser/.luarocks/share/lua/5.4/tl.lua:2479: in upvalue 'parse_statements'
        /home/testuser/.luarocks/share/lua/5.4/tl.lua:2522: in function 'tl.parse_program'
        /home/testuser/.luarocks/share/lua/5.4/tl.lua:7671: in function 'tl.process_string'
        /home/testuser/.luarocks/share/lua/5.4/tl.lua:7634: in function 'tl.process'
        ...user/.luarocks/lib/luarocks/rocks-5.4/tl/0.10.1-1/bin/tl:907: in main chunk
```

**Comment by lenscas** [comment]
just tried it on the playground. The absolute minimum to cause this is
```lua
return (1,) as
```
but, with 1 character more it also crashes (in case that helps)
```lua
return (1,2) as
```

**Comment by euclidianAce** [comment]
Also returning a parenthesized list of expressions is invalid lua, you can only return one expression in parens:
```lua
local function foo()
   return (2, 3)
end
```
```
$ lua foo.lua
lua: foo.lua:3: ')' expected near ','
```
So this should definitely be caught

**Comment by lenscas** [comment]
> Also returning a parenthesized list of expressions is invalid lua, you can only return one expression in parens:

even if it was valid lua, the `as` would definitely make it invalid teal

**Comment by hishamhm** [comment]
Fixed by @euclidianAce in #347 which I merged manually with some minor tweaks!
